src/nbl/ext/FFT/FFT.cpp

```diff
@@ -0,0 +1,368 @@
+﻿// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/ext/FFT/FFT.h"
+#include "../../../../source/Nabla/COpenGLExtensionHandler.h"
+
+#include <cstdio>
+
+using namespace nbl;
+using namespace nbl::asset;
+using namespace nbl::video;
+using namespace ext::FFT;
+
+core::SRange<const asset::SPushConstantRange> FFT::getDefaultPushConstantRanges()
+{
+	static const asset::SPushConstantRange ranges[5] =
+	{
+		{
+			ISpecializedShader::ESS_COMPUTE,
+			0u,
+			sizeof(uint32_t) * 3
+		},
+		{
+			ISpecializedShader::ESS_COMPUTE,
+			sizeof(uint32_t) * 4,
+			sizeof(uint32_t) * 3
+		},
+		{
+			ISpecializedShader::ESS_COMPUTE,
+			sizeof(uint32_t) * 8,
+			sizeof(uint32_t)
+		},
+		{
+			ISpecializedShader::ESS_COMPUTE,
+			sizeof(uint32_t) * 9,
+			sizeof(uint32_t)
+		},
+		{
+			ISpecializedShader::ESS_COMPUTE,
+			sizeof(uint32_t) * 10,
+			sizeof(uint32_t)
+		},
+	};
+	return {ranges, ranges+5};
+}
+
+core::SRange<const video::IGPUDescriptorSetLayout::SBinding> FFT::getDefaultBindings(video::IVideoDriver* driver, DataType inputType)
+{
+	static core::smart_refctd_ptr<IGPUSampler> sampler;
+
+	static IGPUDescriptorSetLayout::SBinding bnd[] =
+	{
+		{
+			0u,
+			EDT_STORAGE_BUFFER,
+			1u,
+			ISpecializedShader::ESS_COMPUTE,
+			&sampler
+		},
+		{
+			1u,
+			EDT_STORAGE_BUFFER,
+			1u,
+			ISpecializedShader::ESS_COMPUTE,
+			nullptr
+		},
+	};
+
+	if (DataType::SSBO == inputType) {
+		bnd[0].type = EDT_STORAGE_BUFFER;
+	} else if (DataType::TEXTURE2D == inputType) {
+		bnd[0].type = EDT_COMBINED_IMAGE_SAMPLER;
+	}
+
+	bnd[0].samplers = &sampler;
+	
+	if (!sampler)
+	{
+		IGPUSampler::SParams params =
+		{
+			{
+				ISampler::ETC_CLAMP_TO_EDGE,
+				ISampler::ETC_CLAMP_TO_EDGE,
+				ISampler::ETC_CLAMP_TO_EDGE,
+				ISampler::ETBC_FLOAT_OPAQUE_BLACK,
```

Comment:
you could set `ETBC_FLOAT_TRANSPARENT_BLACK ` as the border colour and do conditional `fillWithZero ? ETC_CLAMP_TO_BORDER:ETC_CLAMP_TO_EDGE`